scram/route_manager/authentication_backends.py

```diff
@@ -5,48 +5,45 @@
 from mozilla_django_oidc.auth import OIDCAuthenticationBackend
 
 
+def groups_overlap(a, b):
+    """Helper function to see if a and b have any overlap.
+
+    Returns:
+        bool: True if there's any overlap between a and b.
+    """
+    return not set(a).isdisjoint(b)
+
+
 class ESnetAuthBackend(OIDCAuthenticationBackend):
     """Extend the OIDC backend with a custom permission model."""
 
-    def update_groups(self, user, claims):
+    @staticmethod
+    def update_groups(user, claims):
         """Set the user's group(s) to whatever is in the claims."""
-        claimed_groups = claims.get("groups", [])
-
         effective_groups = []
-        is_admin = False
-
-        ro_group = Group.objects.get(name="readonly")
-        rw_group = Group.objects.get(name="readwrite")
-
-        for g in claimed_groups:
-            # If any of the user's groups are in DENIED_GROUPS, deny them and stop processing immediately
-            if g in settings.SCRAM_DENIED_GROUPS:
-                effective_groups = []
-                is_admin = False
-                break
-
-            if g in settings.SCRAM_ADMIN_GROUPS:
-                is_admin = True
-
-            if g in settings.SCRAM_READONLY_GROUPS:
-                if ro_group not in effective_groups:
-                    effective_groups.append(ro_group)
+        claimed_groups = claims.get("groups", [])
 
-            if g in settings.SCRAM_READWRITE_GROUPS:
-                if rw_group not in effective_groups:
-                    effective_groups.append(rw_group)
+        if groups_overlap(claimed_groups, settings.SCRAM_DENIED_GROUPS):
```

Comment:
These changes make sense overall but seem like they don't fit in this MR

Reply:
Same, this was necessary to reduce the complexity of the function in order to get that check to pass.